src/components/charts/ChartGroupUpdate.tsx

```diff
@@ -102,75 +134,213 @@ export default function ChartGroupUpdate({ }) {
-        <>
-            <div className="chart-group--details-page">
-                <div className="page-header">
-                    <div className="flex column left">
-                        <div className="flex left">
-                            <BreadCrumb breadcrumbs={breadcrumbs.slice(1,)} />
-                        </div>
-                        <div className="flex left page-header__title">
-                            {state.name}
-                            <Pencil className="pointer" onClick={e => setChartDetailsUpdate(true)} />
-                        </div>
+    function handleAppStoreChange(event): void {
+        event.preventDefault();
+        let searchParams = new URLSearchParams(location.search);
+        let deprecate = searchParams.get(QueryParams.IncludeDeprecated);
+        let chartRepoId = searchParams.get(QueryParams.ChartRepoId);
+        let qs = `${QueryParams.AppStoreName}=${appStoreName}`;
+        if (deprecate) qs = `${qs}&${QueryParams.IncludeDeprecated}=${deprecate}`;
+        if (chartRepoId) qs = `${qs}&${QueryParams.ChartRepoId}=${chartRepoId}`;
+        history.push(`${url}?${qs}`);
+    }
+
+    function handleDeprecateChange(deprecated): void {
+        let searchParams = new URLSearchParams(location.search);
+        let app = searchParams.get(QueryParams.AppStoreName);
+        let chartRepoId = searchParams.get(QueryParams.ChartRepoId);
+        let qs = `${QueryParams.IncludeDeprecated}=${deprecated}`;
+        if (app) qs = `${qs}&${QueryParams.AppStoreName}=${app}`;
+        if (chartRepoId) qs = `${qs}&${QueryParams.ChartRepoId}=${chartRepoId}`
+        history.push(`${url}?${qs}`);
+    }
+
+    function handleCloseFilter() {
+        setSelectedChartRepo(appliedChartRepoFilter)
+    }
+
+    function handleChartRepoChange(selected): void {
+        let chartRepoId = selected?.map((e) => { return e.value }).join(",");
+        let searchParams = new URLSearchParams(location.search);
+        let app = searchParams.get(QueryParams.AppStoreName);
+        let deprecate = searchParams.get(QueryParams.IncludeDeprecated);
+        let qs = `${QueryParams.ChartRepoId}=${chartRepoId}`;
+        if (app) qs = `${qs}&${QueryParams.AppStoreName}=${app}`;
+        if (deprecate) qs = `${qs}&${QueryParams.IncludeDeprecated}=${deprecate}`;
+        history.push(`${url}?${qs}`);
+    }
+
+
+    function clearSearch(event): void {
+        let searchParams = new URLSearchParams(location.search);
+        let deprecate = searchParams.get(QueryParams.IncludeDeprecated);
+        let chartRepoId = searchParams.get(QueryParams.ChartRepoId);
+        let qs: string = "";
+        if (deprecate) qs = `${qs}&${QueryParams.IncludeDeprecated}=${deprecate}`;
+        if (chartRepoId) qs = `${qs}&${QueryParams.ChartRepoId}=${chartRepoId}`;
+        history.push(`${url}?${qs}`);
+    }
+
+    function initialiseFromQueryParams(chartRepoList) {
+        let searchParams = new URLSearchParams(location.search);
```

Comment:
again these 3-4 lines are repeated every where to get values from query param

Reply:
fixing

Reply:
fixed @pghildiyal

Reply:
fixed